src/ImageSharp/Formats/Ico/IcoFrameMetadata.cs

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) Six Labors.
+// Licensed under the Six Labors Split License.
+
+namespace SixLabors.ImageSharp.Formats.Ico;
+
+/// <summary>
+/// Provides Ico specific metadata information for the image.
+/// </summary>
+public class IcoFrameMetadata : IDeepCloneable
+{
+    /// <summary>
+    /// Initializes a new instance of the <see cref="IcoFrameMetadata"/> class.
+    /// </summary>
+    public IcoFrameMetadata()
+    {
+    }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="IcoFrameMetadata"/> class.
+    /// </summary>
+    /// <param name="width">width</param>
+    /// <param name="height">height</param>
+    /// <param name="colorCount">colorCount</param>
+    /// <param name="planes">planes</param>
+    /// <param name="bitCount">bitCount</param>
+    /// <param name="bytesInRes">bytesInRes</param>
+    /// <param name="imageOffset">imageOffset</param>
+    public IcoFrameMetadata(byte width, byte height, byte colorCount, ushort planes, ushort bitCount, uint bytesInRes, uint imageOffset)
+    {
+        this.Width = width;
+        this.Height = height;
+        this.ColorCount = colorCount;
+        this.Planes = planes;
+        this.BitCount = bitCount;
+        this.BytesInRes = bytesInRes;
+        this.ImageOffset = imageOffset;
+    }
+
+    private IcoFrameMetadata(IcoFrameMetadata metadata)
+    {
+        this.Width = metadata.Width;
+        this.Height = metadata.Height;
+        this.ColorCount = metadata.ColorCount;
+        this.Planes = metadata.Planes;
+        this.BitCount = metadata.BitCount;
+        this.BytesInRes = metadata.BytesInRes;
+        this.ImageOffset = metadata.ImageOffset;
+        this.BmpMetadata = metadata.BmpMetadata;
+        this.PngMetadata = metadata.PngMetadata;
+    }
+
+    /// <summary>
+    /// Gets or sets BmpMetadata.
+    /// </summary>
+    public Bmp.BmpMetadata? BmpMetadata { get; set; }
+
+    /// <summary>
+    /// Gets or sets PngMetadata.
+    /// </summary>
+    public Png.PngMetadata? PngMetadata { get; set; }
+
+    /// <summary>
+    /// Gets or sets Width field. <br />
+    /// Specifies image width in pixels. Can be any number between 0 and 255. Value 0 means image width is 256 pixels.
+    /// </summary>
+    public byte Width { get; set; }
+
+    /// <summary>
+    /// Gets or sets Height field. <br />
+    /// Specifies image height in pixels. Can be any number between 0 and 255. Value 0 means image height is 256 pixels.
+    /// </summary>
+    public byte Height { get; set; }
+
+    /// <summary>
+    /// Gets or sets ColorCount field. <br />
+    /// Specifies number of colors in the color palette. Should be 0 if the image does not use a color palette.
+    /// </summary>
+    public byte ColorCount { get; set; }
```

Comment:
Why not store the palette itself like gif/png?

Reply:
Because the BmpMetadata does not support color palettes yet.
https://github.com/SixLabors/ImageSharp/blob/afe21339afb60061de8a942e1262cac230e6b6a2/src/ImageSharp/Formats/Bmp/BmpMetadata.cs#L41

Reply:
We should store and reuse the palette when available like we do for Gif/Png. It's been on my TODO list for a while. 

https://github.com/SixLabors/ImageSharp/issues/2404

However, this should be a separate PR.